api/src/main.rs

```diff
@@ -15,11 +15,12 @@ mod handlers;
 fn main() -> io::Result<()> {
     env_logger::init();
 
-    HttpServer::new(|| {
-        let db_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
-        let pool = db::init_pool(db_url).expect("Couldn't establish connection to database!");
+    let db_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
+    let pool = db::init_pool(db_url).expect("Couldn't establish connection to database!");
+
+    HttpServer::new(move || {
         App::new()
-            .data(pool)
+            .data(pool.clone())
```

Comment:
can you explain why you need to `clone` it when defined outside the closure?

Reply:
Since it is a closure, the borrow checker cannot know the closure isn't called twice or more. With the `.clone()`, the borrow checker is fine.

Without the `.clone()`, the following error message appears:
```
error[E0507]: cannot move out of captured variable in an `Fn` closure
```